sphinx/screens/dashboard/dashboard/src/main/java/chat/sphinx/dashboard/ui/DashboardFragment.kt

```diff
@@ -287,6 +287,33 @@ internal class DashboardFragment : MotionLayoutFragment<
             }
         }
 
+        binding.layoutDashboardPopup.layoutDashboardPeopleProfilePopup.apply {
+            textViewDashboardPopupClose.setOnClickListener {
+                viewModel.deepLinkPopupViewStateContainer.updateViewState(
+                    DeepLinkPopupViewState.PopupDismissed
+                )
+            }
+
+            buttonSaveProfile.setOnClickListener {
+                onStopSupervisor.scope.launch(viewModel.mainImmediate) {
+                    viewModel.savePeopleProfile()
+                }
+            }
+        }
+
+        binding.layoutDashboardPopup.layoutDashboardDeleteProfilePopup.apply {
+            textViewDashboardPopupClose.setOnClickListener {
+                viewModel.deepLinkPopupViewStateContainer.updateViewState(
+                    DeepLinkPopupViewState.PopupDismissed
+                )
+            }
+            buttonSaveProfile.setOnClickListener {
+                onStopSupervisor.scope.launch(viewModel.mainImmediate) {
+                    viewModel.deletePeopleProfile()
+                }
+            }
+        }
```

Comment:
I think if we move this bit of code to line 525 the delete call will actually call `viewModel.deletePeopleProfile` on DELETE /profile, seems like the onClick is always set to `viewModel.savePeopleProfile` otherwise

But when I'm actually testing this I'm getting a 400 bad request I think we need to pass the host and id in the body of the DELETE call

Reply:
@kevkevinpal thanks for looking into this and testing. I just pushed one commit that basically removed the delete_profile_dashboard_popup layout since I was already changing title and button text for the popup depending on the viewState. Then I'm setting the onClickListener on the save button and defining to call Save or Delete depending on the viewState again. Let me know if the changes make sense.

I'm not sure about the 400 you are getting. I didn't test the DELETE since I didn't find the delete button on people site. And on the JS implementation the Delete call has no params, but I will check with Evan tomorrow.